dlfcn_test.go

```diff
@@ -59,7 +59,9 @@ func buildSharedLib(compilerEnv, libFile string, sources ...string) error {
 	} else {
 		args = []string{"-shared", "-Wall", "-Werror", "-o", libFile}
 	}
-
+	if runtime.GOARCH == "386" {
```

Comment:
If `-m32` indicates a general 32bit architecture, this should be `unsafe.Sizeof(uintptr(0)) == 4`

Reply:
From my understanding it only indicates i386 on amd64 systems. It's listed under the x86_64 options here: https://gcc.gnu.org/onlinedocs/gcc/x86-Options.html